File: back/internal/pkg/domain/model/rtc_remote.go
package model

type RTCRemote struct {
	ConversationsUpdateSeq *UpdateSeqRange `json:"conversations_update_seq"`
	Messages               []MessageRange  `json:"messages"`
	LastKnownUserVersion   UserVersion     `json:"last_known_user_version"`
}

func NewRTCSessionStart() RTCRemote {
	return RTCRemote{
		ConversationsUpdateSeq: nil,
		Messages:               nil,
		LastKnownUserVersion:   UserVersion{},
	}
}

File: app/package/canine_sync/lib/models/rtc_remote.dart
import 'package:freezed_annotation/freezed_annotation.dart';

import 'message_range.dart';
import 'range.dart';
import 'user_version.dart';

part 'rtc_remote.freezed.dart';
part 'rtc_remote.g.dart';

@freezed
class RtcRemote with _$RtcRemote {
  const factory RtcRemote({
    Range? conversationsUpdateSeq,
    List<MessageRange>? messageRange,
    UserVersion? lastKnownUserVersion,
  }) = _RtcRemote;

  factory RtcRemote.fromJson(Map<String, dynamic> json) =>
      _$RtcRemoteFromJson(json);
}
